src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/NullShouldImmediately.java

```diff
@@ -0,0 +1,60 @@
+package com.wiley.autotest.selenium.elements.upgrade.v3;
+
+import com.wiley.autotest.selenium.SeleniumHolder;
+import com.wiley.autotest.selenium.context.SearchStrategy;
+import com.wiley.autotest.selenium.elements.upgrade.TeasyElement;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+
+import java.util.function.Function;
+
+/**
+ * Created by vefimov on 26/04/2017.
+ */
+public class NullShouldImmediately implements Should {
+
+    private TeasyElement element;
+
+    public NullShouldImmediately(TeasyElement element) {
+        this.element = element;
+    }
+
+    //null element can't be displayed
+    public void beDisplayed() {
+        throwException();
+    }
+
+    //null element is absent by default
+    public void beAbsent() {
+        doNothing();
+    }
+
+    public void haveText(String text) {
+        throwException();
+    }
+
+    public void haveAttribute(String attributeName, String value) {
+        throwException();
+    }
+
+    public void haveAttribute(String attributeName) {
+        throwException();
+    }
+
+    public void notHaveAttribute(String attributeName) {
+        throwException();
+    }
+
+    public void customCondition(Function<WebDriver, Boolean> condition) {
+        new TeasyFluentWait<>(SeleniumHolder.getWebDriver(), new SearchStrategy(0)).waitFor(condition);
+    }
+
+    private void throwException() {
```

Comment:
What if have default implementation of this method?

Reply:
Thought about it...then it will become a public method of an interface which it should not be... it's not related to a Should anyhow. The only solution is to make Should abstract class... will think about it later...